### DISCUSSION TITLE
Configuring maxDuration in vercel.json is invalid when 504 time out.

### DISCUSSION THREAD

**Discussion by huoyingjianshen** [vercel/community#7070] [Bug Report]
### Page to Investigate

Task timed out after 15.02 seconds

### Steps to Reproduce

I am a pro user. I configured the maxDuration of my serverless function to 300 in vercel.json according to the configuration information in the document, but "Task timed out after 15.02 seconds" still appeared. Could you tell me how this works and how I use app route? Does it matter? For example, my configuration is: "{
  "functions": {
    "app/api/v1/xxx/route.ts": {
      "maxDuration": 300
    }
  }
}"

**Comment by swarnava** [comment]
Hi,

What framework are you using? Have you tried setting up within Serverless function it self https://vercel.com/docs/functions/configuring-functions/duration#node.js-next.js-%3E=-13.5-or-higher-sveltekit-astro-nuxt-and-remix?
